sklearn/preprocessing/tests/test_data.py

```diff
@@ -2003,6 +2006,19 @@ def test_binarizer(constructor):
             binarizer.transform(constructor(X))
 
 
+@pytest.mark.parametrize(
+    "array_namespace, device, _", yield_namespace_device_dtype_combinations()
+)
+def test_binarizer_array_api_int(array_namespace, device, _):
+    xp = _array_api_for_tests(array_namespace, device)
+    X_np = np.reshape(np.asarray([0, 1, 2, 3, 4]), (-1, 1))
```

Comment:
I think this test should be expanded to test both on `X_np` with `dtype=np.int32` and with `dtype_name` for a floating point dtype.

Reply:
Isn't `dtype_name` tested in `test_preprocessing_array_api_compliance`, where I added an instance of `Binarizer`?